src/scenes/Partnerships/Edit/EditPartnership.tsx

```diff
@@ -102,19 +74,20 @@ export const EditPartnership: FC<EditPartnershipProps> = ({
   );
 
   return (
-    <DialogForm<UpdatePartnershipInput & SubmitAction<'delete'>>
+    <PartnershipForm<EditPartnershipFormInput>
       {...props}
-      onSubmit={async (input) => {
+      onlyDirtySubmit={false} // Lets us delete without changing any fields
```

Comment:
Heck. Maybe we should also allow a string here for to limit to this functionality to the specified submit actions?

Reply:
So maybe change the prop from `onlyDirtySubmit` to something like `allowCleanSubmit: 'never' | 'always' | 'delete'`, and default would be `never`?

Reply:
I do like changing the prop name so the default is false. I like "dirty" -> "clean", but "allow" feels wrong. We "allow" clean submits, they just do nothing.

Submit action names are dynamic though; they can be whatever string you specify. So that's why I was thinking of using a boolean to distinguish between a submit action or applying it to all.

Reply:
Okay, so you're just thinking the type of `onlyDirtySubmit` would be `boolean | string | undefined`, default to `false`, and we just check for a string value and compare to `input.SubmitAction`?

I'm not married to `allowCleanSubmit`, but to argue against your objection, `onlyDirtySubmit` implies a similar (opposite) thing: that we don't "allow" dirty submits

Reply:
So the `onSubmit` prop of `DialogForm` would look like this:

```
onSubmit={async (data, form) => {
    if (onlyDirtySubmit === true && !form.getState().dirty) {
      onClose?.('cleanSubmit', form);
      return null;
    }
    if (
      !onlyDirtySubmit ||
      onlyDirtySubmit === (data as any).submitAction
    ) {
      try {
        const res = await onSubmit(data, form);
        onSuccess?.(res);
        onClose?.('success', form);
        return null;
      } catch (e) {
        return await handleFormError(e, errorHandlers);
      }
    }
  }}
```

Reply:
> I'm not married to `allowCleanSubmit`, but to argue against your objection, `onlyDirtySubmit` implies a similar (opposite) thing: that we don't "allow" dirty submits

Yeah I agree. Maybe we just remove the option all together. I can't even think of a use case where we want to send clean submissions. If we know it's clean, there's no point. Actually scratch that - delete is a use case 🤦 . What about `sendCleanSubmissions`?
I still like changing it to "clean" because those are the exceptions. Maybe "send" is the verb we are looking for, or something similar.

Reply:
> What about `sendCleanSubmissions`?

Sounds good to me!

Reply:
So the default then would be false, and any strings given that match the submit action would call onSubmit even if clean (skip the dirty check).

Change your mind on `sendIfClean`? I could get behind that too.

Reply:
I didn't as much change my mind as decide it wasn't worth haggling over it. Up to you

Reply:
Haha I'm feeling the same way. I'm good with anything like that

Reply:
`sendIfClean` it is! 'Cuz it's shorter

Reply:
Could you update the prop description as well?

Reply:
🤪

Reply:
@CarsonF Ready again